datadog_lambda/dsm.py

```diff
@@ -0,0 +1,37 @@
+from datadog_lambda import logger
+from datadog_lambda.trigger import EventTypes
+
+
+def set_dsm_context(event, event_source):
+
+    if event_source.equals(EventTypes.SQS):
+        _dsm_set_sqs_context(event)
+
+
+def _dsm_set_sqs_context(event):
+    from datadog_lambda.wrapper import format_err_with_traceback
+
+    from ddtrace.internal.datastreams.processor import (
+        DataStreamsProcessor as processor,
+        DsmPathwayCodec,
+    )
+    from ddtrace.internal.datastreams.botocore import (
+        get_datastreams_context,
+        calculate_sqs_payload_size,
+    )
+
+    records = event.get("Records", [])
```

Comment:
nit: even if a `"Records"` key is in the `event` dict, this will still allocate an empty list. Try this instead

```suggestion
    records = event.get("Records") or []
```

which only allocates the empty list when there are no records.

Though, even better yet, why allocate an empty list at all when you could just do

```python
records = event.get("Records")
if records is None:
    return
for record in records:
    ... do stuff ...
```